x-pack/elastic-agent/pkg/core/logger/logger.go

```diff
@@ -127,6 +130,20 @@ func makeInternalFileOutput(cfg *Config) (zapcore.Core, error) {
 		return nil, errors.New("failed to create internal file rotator")
 	}
 
-	encoder := zapcore.NewJSONEncoder(ecszap.ECSCompatibleEncoderConfig(logp.JSONEncoderConfig()))
+	encoderConfig := ecszap.ECSCompatibleEncoderConfig(logp.JSONEncoderConfig())
+	encoderConfig.EncodeTime = utcTimestampEncode
+	encoder := zapcore.NewJSONEncoder(encoderConfig)
 	return ecszap.WrapCore(zapcore.NewCore(encoder, rotator, cfg.Level.ZapLevel())), nil
 }
+
+// utcTimestampEncode is a zapcore.TimeEncoder that formats time.Time in ISO-8601 in UTC.
```

Comment:
`logp.JSONEncoderConfig` uses zap's ISO8601 encoder; this function copies that pattern but forces the time into UTC

Reply:
looks ok, i will test it a bit later